x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.test.ts

```diff
@@ -334,9 +356,9 @@ describe.each([
               ],
             },
           ],
+          results: getResultsStructure(),
         },
         message: 'Bulk edit partially failed',
-        status_code: 500,
```

Comment:
> Replaced the usage of response.custom for error response with response.customError to be consistent with other Kibana plugins. This introduces a slight breaking change as response.customError doesn't pass the status code to the response body.

can we actually introduce a breaking change in minor release?

Reply:
I remember, including `status_code: 500` to final error response was also done to keep it in line with SecuritySolution error responses, that uses this property as part of error response.

 @banderror, @spong , what are your thoughts? Is it safe to introduce such change in minor release?

Reply:
If you're talking about `SiemResponseFactory`, it's pretty limited and looks like a relic of the past to me. For example, it accepts a `body` param, which is actually a `message` in disguise; it doesn't allow for additional error attributes to be passed, and its schema doesn't comply with the core framework's error responses. 

So, in my opinion, we should gradually migrate away from the `SiemResponseFactory` to the standard core functionality.

Reply:
I agree on that, but not sure this is something we have to do before BC, introducing breaking change to error response schema as well

Reply:
I think I'm in agreement with @xcrzx -- [SiemResponseFactory](https://github.com/elastic/kibana/blame/ba0833f42b89a0b62bccd1d8c04bf9c799f32258/x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts#L158-L181) is almost a couple years old now (so would be good to review its uses/abstraction), and it has been copied elsewhere like `lists` as well ([source](https://github.com/elastic/kibana/blob/6d80bf7a217dafc2740c0f8a4e75cf31081e8d45/x-pack/plugins/lists/server/routes/utils/build_siem_response.ts#L43-L73)), so we may want to check with the @elastic/security-solution-platform folks and see if they have any plans for it.

As for if this is an API breaking change, I think that line is blurred. Sure we don't [document the error structure ](https://www.elastic.co/guide/en/security/current/bulk-actions-rules-api.html#_response_payload_8), but it is a documented API, and I think it's fair to say consumers of the API are going to end up relying on the response error structure (and perhaps that's just a release notes blurb that it's been updated?). 

That said, if we're okay with this change, it shouldn't be an issue merging post-FF as we're still in the early BC's and this is a follow-up fix PR finalizing the updated bulk actions routes.

Reply:
>  Sure we don't [document the error structure ](https://www.elastic.co/guide/en/security/current/bulk-actions-rules-api.html#_response_payload_8), but it is a documented API, and I think it's fair to say consumers of the API are going to end up relying on the response error structure (

Yeah, that's my concern too.

Reply:
I don't think we should introduce any breaking changes in this PR. Thoughts here: https://github.com/elastic/kibana/pull/125551#discussion_r807789107